pkgs/development/compilers/dotnet/default.nix

```diff
@@ -30,16 +29,24 @@ let
 
   # Convert a "stdenv.hostPlatform.system" to a dotnet RID
   systemToDotnetRid = system: runtimeIdentifierMap.${system} or (throw "unsupported platform ${system}");
-in
-{
-  inherit systemToDotnetRid;
-
-  combinePackages = attrs: callPackage (import ./combine-packages.nix attrs) {};
-} // lib.optionalAttrs config.allowAliases {
-  # EOL
-  sdk_2_1 = throw "Dotnet SDK 2.1 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_2_2 = throw "Dotnet SDK 2.2 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_3_0 = throw "Dotnet SDK 3.0 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_3_1 = throw "Dotnet SDK 3.1 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-  sdk_5_0 = throw "Dotnet SDK 5.0 is EOL, please use 6.0 (LTS) or 7.0 (Current)";
-} // dotnet_6_0 // dotnet_7_0 // dotnet_8_0
+
+  self = {
+    inherit systemToDotnetRid;
+
+    combinePackages = attrs: callPackage (import ./combine-packages.nix attrs) {};
```

Comment:
I believe something we can do here is `withExtraSdks` and then imply the dotnet that this is part of as the CLI (first arg).

e.g.: `dotnetPackages_9.withExtraSdks [dotnetPackages_8.sdk]` would be the same as `dotnetPackages_9.combinePackages [dotnetPackages_9.sdk dotnetPackages_8.sdk]`.

Reply:
Done in latest.